### DISCUSSION TITLE
api_response not returning full output

### DISCUSSION THREAD

**Discussion by zactuscactus** [NREL/REopt-API-Analysis#128] [8 - API: Application Programming Interface]
Hello, I have noticed that the api seems to not return the full set of outputs from optimization I am running. For example, I ran an optimization that returns financial outcomes such as capital investments (see the screenshot below), but it does not return what the investments were (e.g. the PV and ElectricStorage fields are not present). It is also missing the 'Site' output, where carbon reduction would be. 
I am sure I am missing something, but it is unclear what that could be.
![image](https://github.com/NREL/REopt-Analysis-Scripts/assets/16196121/aa4bca4d-05d2-4a52-8fd0-cef5c2a10db7)

**Comment by adfarth** [comment]
Apologies for the delayed response--this is unexpected! Has this issue persisted for you? If so, could you share the post that returned this result?

**Reply by zactuscactus** [reply]
Hey, thanks so much for the response!! See an attached example.
[results_file_site_54.json](https://github.com/NREL/REopt-Analysis-Scripts/files/13754425/results_file_site_54.json)

See below the store costs, but no response for storage or emissions.
![image](https://github.com/NREL/REopt-Analysis-Scripts/assets/16196121/0fa6e278-8d25-4525-94bd-2e4c5e00ca90)

Also, quick follow up question. Where can I find the rate limit for optimization requests? I have scoured the docs, but don't see it mentioned explicitly.

**Reply by adfarth** [reply]
I just re-ran your analysis using the example code in this repository (https://github.com/NREL/REopt-Analysis-Scripts/blob/master/REopt%20API%20Scripts/single_scenario_example.ipynb).  Here is the results json (which surprisingly has very different results from what you shared, but does contain all expected output keys):
[results_file_test.json](https://github.com/NREL/REopt-Analysis-Scripts/files/13754702/results_file_test.json).

I used the "inputs" section of the results json that you shared to run this scenario. I.e.: 
```
# Load a json into a python dictionary
load_post = "results_file_site_54"
with open(os.path.join(inputs_path, load_post + ".json"), 'r') as fp:
    post_1 = json.load(fp)["inputs"]

outputs_file_name = "results_file_test"
root_url = "https://developer.nrel.gov/api/reopt/stable" # /stable == /v3 
api_response = get_api_results(post=post_1, 
                               API_KEY=API_KEY, 
                               api_url=root_url, 
                               results_file=os.path.join(outputs_path, outputs_file_name + ".json"), 
                               run_id=None)
```

Could you try re-running in this way and confirm if the api_response contains all expected keys? If it does but the results are not as expected, perhaps you could share the original `post` that you sent.
